domain/services/model.go

```diff
@@ -138,24 +140,32 @@ func NewModelServices(
 			},
 			modelDB: modelDB,
 		},
-		modelUUID:              modelUUID,
-		providerFactory:        providerFactory,
-		modelObjectStoreGetter: modelObjectStoreGetter,
-		storageRegistry:        storageRegistry,
-		publicKeyImporter:      publicKeyImporter,
-		leaseManager:           leaseManager,
-		logDir:                 logDir,
-		clock:                  clock,
+		modelUUID:                   modelUUID,
+		providerFactory:             providerFactory,
+		controllerObjectStoreGetter: controllerObjectStoreGetter,
+		modelObjectStoreGetter:      modelObjectStoreGetter,
+		storageRegistry:             storageRegistry,
+		publicKeyImporter:           publicKeyImporter,
+		leaseManager:                leaseManager,
+		logDir:                      logDir,
+		clock:                       clock,
 	}
 }
 
 // AgentBinaryStore returns the model's [agentbinaryservice.AgentBinaryStore]
 // for the current model.
 func (s *ModelServices) AgentBinaryStore() *agentbinaryservice.AgentBinaryStore {
+	controllerAgentBinaryStore := agentbinaryservice.NewAgentBinaryStore(
+		agentbinarystate.NewState(changestream.NewTxnRunnerFactory(s.controllerDB)),
+		s.logger.Child("modelagentbinary"),
+		s.controllerObjectStoreGetter,
+		nil,
+	)
 	return agentbinaryservice.NewAgentBinaryStore(
```

Comment:
Please tell me we've got an alternative coming for this.

Reply:
I've had a chat with @tlm to understand the original intent and I'm removing this. We'll just fallback to the controller AgentBinaryStore in the facade.